core/trino-main/src/main/java/io/trino/sql/planner/plan/TableScanNode.java

```diff
@@ -93,6 +98,26 @@ public TableScanNode(
         checkArgument(assignments.keySet().containsAll(outputs), "assignments does not cover all of outputs");
         this.enforcedConstraint = requireNonNull(enforcedConstraint, "enforcedConstraint is null");
         this.forDelete = forDelete;
+
+        if (!enforcedConstraint.isAll() && !enforcedConstraint.isNone()) {
```

Comment:
nit: can we move this to a checkNoDanglingColumns function?

Reply:
i prefer not to call functions within constructor (in general at least), so if there isn't a compelling reason to do so, would prefer to keep it here

Reply:
sure, i can do this, if you feel strongly about it.
i'd prefer to keep as is, though

Reply:
yea let's do it. I usually prefer that than having to leave more comments, which this would need without that function context)